Sources/SpeziOnboarding/OnboardingFlow/OnboardingNavigationPath.swift

```diff
@@ -97,30 +120,180 @@ public class OnboardingNavigationPath {
     /// In case there isn't a suitable element within the `OnboardingNavigationPath.path`, return the `OnboardingStepIdentifier`
     /// of the first onboarding view.
     private var currentOnboardingStep: OnboardingStepIdentifier? {
-        guard let lastElement = path.last(where: { !$0.custom }) else {
+        guard let lastElement = path.last(where: { !$0.isCustom }) else {
             return firstOnboardingStepIdentifier
         }
-
         return lastElement
     }
-
+    
+    /// Creates an empty, unconfigured `OnboardingNavigationPath`.
+    ///
+    /// This initializer is intended for creating empty, unconfigured `OnboardingNavigationPaths` which are then injected into an ``OnboardingStack``.
+    public init() {}
+    
+    
     /// An `OnboardingNavigationPath` represents the current navigation path within the ``OnboardingStack``.
     /// - Parameters:
     ///   - views: SwiftUI `View`s that are declared within the ``OnboardingStack``.
-    ///   - complete: An optional SwiftUI `Binding` that is injected by the ``OnboardingStack``.
+    ///   - isComplete: An optional SwiftUI `Binding` that is injected by the ``OnboardingStack``.
     ///     Is managed by the ``OnboardingNavigationPath`` to indicate whether the onboarding flow is complete.
-    ///   - startAtStep: An optional SwiftUI (Onboarding) `View` type indicating the first to-be-shown step of the onboarding flow.
-    init(views: [any View], complete: Binding<Bool>?, startAtStep: (any View.Type)?) {
-        self.complete = complete
-        updateViews(with: views)
-
+    ///   - startAtStep: Optionally, the step the OnboardingNavigationPath should initially move to.
+    func configure(elements: [_OnboardingFlowViewCollection.Element], isComplete: Binding<Bool>?, startAtStep: StepReference?) {
+        didConfigure = true
+        self.isComplete = isComplete
+        updateViews(with: elements)
         // If specified, navigate to the first to-be-shown onboarding step
-        if let startAtStep {
-            append(startAtStep)
+        switch startAtStep {
+        case nil:
+            break
+        case .viewType(let viewType):
+            appendStep(viewType)
+        case .identifier(let hashable):
+            appendStep(hashable)
         }
     }
+    
+    /// Internal function used to update the onboarding steps within the ``OnboardingNavigationPath`` if the
+    /// result builder associated with the ``OnboardingStack`` is reevaluated.
+    ///
+    /// This may be the case with `async` properties that are stored as a SwiftUI `State` in the respective view.
+    ///
+    /// - Parameters:
+    ///   - views: The updated `View`s from the ``OnboardingStack``.
+    func updateViews(with elements: [_OnboardingFlowViewCollection.Element]) {
```

Comment:
In general: I didnt think through everything in here, seems to be quite complex code, still some comments & TODO left.

Reply:
this is one of the things we should revisit in depth when moving the OnboardingStack into eg SpeziViews: IMO the restriction of only taking changs to upcoming steps into account is overly limiting. I tried to sketch out a simple alternative implementation that would also apply changes to already-pushed views but that didn't work properly (we could probably make it work somehow but i didn't investigate enough).

For now i've moved mostly back to the previous implementation here, with some adjustments and additional correctness checks

Reply:
Why would the restriction of only taking changes to upcoming steps be limiting? Do you have a specific use case in mind?
If you enable to modify the Stack's already "passed" Views, you'll get into tons of weird behavior for the user where e.g. Views disappear when navigating back as a certain condition is now set to the opposite. Ofc, there are probably ways to mitigate that, but back then we didn't have the use case for it and I'm also not seeing the use case right now.

But I'm very open to your ideas if there's a good and relatively straightforward solution, feel free to chat me up!